src/datasources/asset/AssetDataSource.ts

```diff
@@ -82,4 +87,15 @@ export class AssetDataSource extends DataSourceBase<AssetQuery, AssetDataSourceO
   getListAssetsSource(): ListAssetsDataSource {
     return this.listAssetsDataSource;
   }
+
+  async metricFindQuery(query: AssetQuery, options: LegacyMetricFindQueryOptions): Promise<MetricFindValue[]> {
```

Comment:
I would recommend defining a separate interface for the variable query like we do here: https://github.com/ni/systemlink-grafana-plugins/blob/a315a292ff6576996bbf89ea2a9fc1c912678870/src/datasources/system/SystemDataSource.ts#L84